sql/core/src/test/scala/org/apache/spark/sql/streaming/test/DataStreamReaderWriterSuite.scala

```diff
@@ -535,15 +536,15 @@ class DataStreamReaderWriterSuite extends StreamTest with BeforeAndAfter with Pr
 
   test("MemorySink can recover from a checkpoint in Complete Mode") {
     val checkpointLoc = newMetadataDir
-    val checkpointDir = new File(checkpointLoc, "offsets")
+    val checkpointDir = new File(new URI(s"$checkpointLoc/offsets"))
```

Comment:
I don't really have a better idea, but this looked a little funny. A URI isn't generally going to be interpreted correctly as a local file path unless it happens to not contain a scheme. Does is something like `new URI(...).getPath` better?